packages/ssr-web/app/services/app-context.ts

```diff
@@ -46,8 +64,12 @@ export default class AppContext extends Service implements AppContextService {
 
   get cardSpaceId() {
     if (this.isCardSpace) {
-      let id = this.host.replace(this.hostSuffixPattern, '') ?? '';
-      return id;
+      if (this.queryIsCardSpace) {
+        return cardSpaceSlugQueryRegex.exec(window.location.search)![1];
+      } else {
+        let id = this.host.replace(this.hostSuffixPattern, '') ?? '';
+        return id;
+      }
```

Comment:
Before I prepare this for review, thoughts on testability for these, @Aierie? I’m reluctant to add tests for this narrow use case but also this hackery is pretty uggers

Reply:
i think this is ok, since this is what the service was created for, and we have a test that can be extended to cover more tricky cases.